File: 2956-find-common-elements-between-two-arrays/2956-find-common-elements-between-two-arrays.java
class Solution {
    public int[] findIntersectionValues(int[] nums1, int[] nums2) {
        int n1 = nums1.length;
        int n2 = nums2.length;
        HashSet<Integer> m1 = new HashSet<>();
        HashSet<Integer> m2 = new HashSet<>();
        for (int i:nums1){
            m1.add(i);
        }
        for (int i:nums2){
            m2.add(i);
        }
        int left = 0,right = 0;
        for (int i = 0;i<n1;i++){
            if (m2.contains(nums1[i])){
                left++;
            }
        }
        for (int i = 0;i<n2;i++){
            if (m1.contains(nums2[i])){
                right++;
            }
        }
        return new int[]{left,right};
    }
}

File: 2956-find-common-elements-between-two-arrays/2956-find-common-elements-between-two-arrays.py
class Solution(object):
    def findIntersectionValues(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        arr = []
        set1 = set()
        set2 = set()
        for i in nums1:
            set1.add(i)
        for j in nums2:
            set2.add(j)
        left = 0
        right = 0
        for i in nums1:
            if (i in set2):
                left = left + 1
        for i in nums2:
            if i in set1:
                right = right+1  
        arr.append(left)
        arr.append(right)
        return arr
        